nsv13/code/modules/overmap/fighters/fighters.dm

```diff
@@ -0,0 +1,355 @@
+				to_chat(user, "<span class='notice'>You pry open the maintenance hatch on [src].</span>")
+				maint_state = MS_OPEN
+				return TRUE
+		if(MS_OPEN)
+			to_chat(user, "<span class='notice'>You start replace the maintenance hatch on [src]...</span>")
+			if(tool.use_tool(src, user, 40, volume=100))
+				to_chat(user, "<span class='notice'>You replace the maintenance hatch on [src].</span>")
+				maint_state = MS_UNSECURE
+				return TRUE
+
+/obj/structure/overmap/fighter/welder_act(mob/user, obj/item/tool)
+	. = FALSE
+	if(obj_integrity/max_integrity*100 >= 100 && user.a_intent != INTENT_HARM)
+		to_chat(user, "<span class='notice'>[src] isn't in need of repairs.</span>")
+		return TRUE
+	else if(maint_state != MS_OPEN && obj_integrity/max_integrity*100 >= 51 && user.a_intent != INTENT_HARM)
+		to_chat(user, "<span class='notice'>You start welding some dents out of [src]'s hull...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>You weld some dents out of [src]'s hull.</span>")
+			obj_integrity += min(10, max_integrity-obj_integrity)
+			if(obj_integrity == max_integrity)
+				to_chat(user, "<span class='notice'>[src] is fully repaired.</span>")
+			return TRUE
+	else if(maint_state == MS_OPEN && obj_integrity/max_integrity*100 >= 51 && user.a_intent != INTENT_HARM)
+		to_chat(user, "<span class='notice'>You need to close [src]'s maintenance panel to begin hull repairs.</span>")
+		return TRUE
+	else if(maint_state == MS_OPEN && obj_integrity/max_integrity*100 <= 50 && user.a_intent != INTENT_HARM)
+		to_chat(user, "<span class='notice'>You start repairing the inner components of [src]...</span>")
+		if(tool.use_tool(src, user, 40, volume=100))
+			to_chat(user, "<span class='notice'>You repair the inner components of [src].</span>")
+			obj_integrity += min(10, max_integrity-obj_integrity)
+			if(obj_integrity == max_integrity)
+				to_chat(user, "<span class='notice'>[src]'s inner components are fully repaired.</span>")
+			return TRUE
+	else if(maint_state == MS_OPEN && obj_integrity/max_integrity*100 <= 50 && user.a_intent != INTENT_HARM)
+		to_chat(user, "<span class='notice'>You open the maintenance panel to repair the inner components of [src].</span>")
+		return TRUE
+
+
+/obj/structure/overmap/fighter/MouseDrop_T(obj/structure/A, mob/user)
+	. = ..()
+	if(istype(A, /obj/structure/munition))
+		switch(maint_state)
+			if(!MS_OPEN)
+				to_chat(user, "<span class='notice'>You require [src] to be in maintenance mode to load munitions!.</span>")
+				return
+			if(MS_OPEN)
+				var/tp = 0
+				for(var/obj/structure/munition/mu in contents)
+					tp++
+				if(tp < 2)
+					to_chat(user, "<span class='notice'>You start adding [A] to [src]...</span>")
+					if(!do_after(user, 2 SECONDS, target=src))
+						return
+					to_chat(user, "<span class='notice'>You add [A] to [src].</span>")
+					A.forceMove(src)
+					playsound(src, 'nsv13/sound/effects/ship/mac_load.ogg', 100, 1)  //placeholder

```

Comment:
Note to self:
Add torpedo firing backend on the fighter

Reply:
I kind of assumed you'd want to stick a couple of torps on fighters as heavy weapons, and do space 80s gun duels for most of the other combat.

Reply:
Yeah that's the plan, but I would like them to have a couple

Reply:
There is currently room for 2 sitting in the "internal" bay, it can be expanded to whatever number you want.